src/write.c

```diff
@@ -300,12 +300,12 @@ static void avifEncoderWriteTrackMetaBox(avifEncoder * encoder, avifRWStream * s
             contentSize = (uint32_t)item->encodeOutput->samples.sample[0].data.size;
         }
 
-        avifRWStreamWriteU16(s, item->id);              // unsigned int(16) item_ID;
-        avifRWStreamWriteU16(s, 0);                     // unsigned int(16) data_reference_index;
-        avifRWStreamWriteU16(s, 1);                     // unsigned int(16) extent_count;
-        avifEncoderItemAddMdatFixup(item, s);           //
-        avifRWStreamWriteU32(s, 0 /* set later */);     // unsigned int(offset_size*8) extent_offset;
-        avifRWStreamWriteU32(s, (uint32_t)contentSize); // unsigned int(length_size*8) extent_length;
+        avifRWStreamWriteU16(s, item->id);          // unsigned int(16) item_ID;
+        avifRWStreamWriteU16(s, 0);                 // unsigned int(16) data_reference_index;
+        avifRWStreamWriteU16(s, 1);                 // unsigned int(16) extent_count;
+        avifEncoderItemAddMdatFixup(item, s);       //
+        avifRWStreamWriteU32(s, 0 /* set later */); // unsigned int(offset_size*8) extent_offset;
+        avifRWStreamWriteU32(s, contentSize);       // unsigned int(length_size*8) extent_length;
```

Comment:
I removed the (uint32_t) cast for contentSize because contentSize is of the uint32_t type.